### PR TITLE
display data unit in html repr

### PR DESCRIPTION
**Pull request by jiaqing23**
## PR Description

xref #5388, data unit into html repr of `sunpy.map.Map`

|image|image|
|-|-|

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -401,7 +401,7 @@ def _repr_html_(self, compute_dask=False):
         ax.semilogy()
         # Explicitly set the power limits for the X axis formatter to avoid text overlaps
         ax.xaxis.get_major_formatter().set_powerlimits((-3, 4))
-        ax.set_xlabel('Pixel value in linear bins')
+        ax.set_xlabel(f'Pixel value{" (" + str(self.unit) +") " if self.unit is not None else " "}in linear bins')
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:404]: Can we not tidy this string up a bit?

Comment (jiaqing23) [reply sunpy/map/mapbase.py:404]: Was thinking to create an temp variable:
```
display_unit = f"({str(self.unit)})" if self.unit is not None else " "
ax.set_xlabel('Pixel value' + display_unit + 'in linear bins') 
```

Comment (nabobalis) [reply sunpy/map/mapbase.py:404]: That sounds worse?

Comment (jiaqing23) [reply sunpy/map/mapbase.py:404]: This looks better:
```
display_unit = f' ({self.unit})' if self.unit is not None else ''
ax.set_xlabel(f'Pixel value{unit_string} in linear bins')
```

Comment (nabobalis) [reply sunpy/map/mapbase.py:404]: I don't think we need a variable for this.

Comment (jiaqing23) [reply sunpy/map/mapbase.py:404]: Shorten the code: 

ax.set_xlabel(f"Pixel value{' (' + self.unit + ')' if self.unit else ''} in linear bins")

Comment (nabobalis) [reply sunpy/map/mapbase.py:404]: Sure

Comment (jiaqing23) [reply sunpy/map/mapbase.py]: Updated the code

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks @jiaqing23

**Pull request merged**